permission/permission.go

```diff
@@ -3,30 +3,40 @@ package permission
 import (
 	"context"
 	"github.com/jmoiron/sqlx"
-	"github.com/ystv/web-auth/role"
 )
 
 type (
 	// Repo where all user data is stored
 	Repo interface {
 		GetPermissions(ctx context.Context) ([]Permission, error)
-		GetPermissionsForRole(ctx context.Context, r role.Role) ([]Permission, error)
 		GetPermission(ctx context.Context, p Permission) (Permission, error)
-		AddPermission(ctx context.Context, p Permission) (Permission, error)
+		AddPermission(ctx context.Context, p1 Permission) (Permission, error)
+		EditPermission(ctx context.Context, p1 Permission) (Permission, error)
+		DeletePermission(ctx context.Context, p1 Permission) error
+
+		getPermissions(ctx context.Context) ([]Permission, error)
+		getPermission(ctx context.Context, p1 Permission) (Permission, error)
+		addPermission(ctx context.Context, p1 Permission) (Permission, error)
+		editPermission(ctx context.Context, p1 Permission) (Permission, error)
+		deletePermission(ctx context.Context, p1 Permission) error
```

Comment:
Quick navigation? They can go if you prefer

Reply:
I still don't understand what they're for and why there's both an exported and unexported copy of each one, especially when the exported ones seemingly just delegate to the unexported ones.

Furthermore, there's not much point in having an unexported method on an interface, because then nobody outside your package can implement it - though that's partly a wider point about these interfaces being slightly redundant in their current state.

Reply:
I can remove them in that case, they don't serve very much purpose

Reply:
That may be better, otherwise a reader thinks there's a purpose for there being two different methods when there really isn't.

Reply:
As in all of the interfaces or just the unexported methods, either is fine with me

Reply:
The interfaces is a wider discussion, but the unexported ones can definitely go.

Reply:
Removed unexported for now and will resolve this for now, we'll discuss more about them as a whole